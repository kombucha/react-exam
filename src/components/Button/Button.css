.Button {
  display: inline-block;
  border: none;
  background: none;
  color: #feda4a;
  font-family: inherit;
  font-size: 2rem;
  cursor: pointer;
  opacity: 0.9;
}

.Button:hover {
  opacity: 1;
}

.Button:disabled {
  opacity: 0.5;
  cursor: default;
}
