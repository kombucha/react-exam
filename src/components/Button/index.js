import React from "react";
import "./Button.css";

const Button = ({ className = "", ...props }) => (
  <button className={`Button ${className}`} {...props} />
);

export default Button;
