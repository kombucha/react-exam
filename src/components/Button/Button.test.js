import React from "react";
import Button from "./index";
import { shallow } from "enzyme";

it("should render properly", () => {
  const wrapper = shallow(<Button />);
  expect(wrapper).toMatchSnapshot();
});
