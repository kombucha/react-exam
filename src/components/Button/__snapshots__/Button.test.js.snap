// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "className": "Button ",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": "Button ",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
